src/data/matrix/basic.lean

```diff
@@ -1232,6 +1232,24 @@ begin
   { rwa [update_column_ne h, if_neg h] }
 end
 
+@[simp] lemma update_column_subsingleton [subsingleton m] (A : matrix n m R)
+  (i : m) (b : n → R) :
+  A.update_column i b = λ i _, b i :=
```

Comment:
Is the RHS better as something like `(col b).reindex (function.const unit.star) id`?